keps/sig-instrumentation/1933-secret-logging-static-analysis/README.md

```diff
@@ -0,0 +1,586 @@
+
+Taint propagation analysis gives additional consideration is given to how data
+ may be transformed or extracted to other values via *propagators*,
+ and how source data may be processed for safe consumption via *sanitizers*.
+See the `go-flow-levee` documentation for details.
+
+While configuration of source identification can be done via manually configured regexp,
+this KEP would benefit from a set of standard Kubernetes go lang struct tags indicating which fields are expected to contain secrets,
+as proposed in [KEP-1753](https://github.com/kubernetes/enhancements/pull/1754).
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+`go-flow-levee` is in active development and has not yet reached a `v1.0` release.
+It is, however, being developed with specific interest in application to Kubernetes.
+
+
+### Risks and Mitigations
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+As a blocking test, there is a risk for developer toil in the event of any
+false-positive or test flakiness.
+This can be mitigated by any contributed with `/override` permissions.
+
+Similarly, depending the order in which PRs are tested and merged,
+ it is theoretically possible that a merge would result in a violation reaching master.
+If this is not properly handled, all PRs made at that time could present as
+ failing analysis.
+This can be mitigated by consistent scans of `master` as a baseline.
+
+As this analysis depends on project-specific considerations of what constitutes
+ a secret or a sink, periodic review is required to ensure configuration is kept up-to-date.
+This is mitigated somewhat with a consistent use of field tags,
+ as proposed in [KEP-1753](https://github.com/kubernetes/enhancements/pull/1754),
+ though correct application of field tags would also be subject to periodic review.
+
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Similar to [KEP-20190605: Metrics Validation and Verification](keps/sig-instrumentation/20190605-metrics-validation-and-verification.md),
+this KEP would introduce a test build target to execute analysis during testing.
+The target will be introduced to `//test/instrumentation`, with configuration data 
```

Comment:
linters and bazel don't mix very well, you can't really cache something that consumes the whole source tree.
we already have an entrypoint for running analysis tools / linters in the verify presubmits (under make) and precedent for managing this sort of tooling in hack/tools/go.mod without pulling it into the main vendor.

I don't think test/instrumentation is the right place for this? This should probably be under hack/

Reply:
Hmmm... I'm glad you commented.  I learned this the "fun" way while doing the proof-of-concept.  I must've either dropped an edit here or forgotten to update this section.  Correction forthcoming.

> I don't think test/instrumentation is the right place for this? This should probably be under hack/

I hadn't known about that collection of tools.  It's absolutely a better place for this.  👍